sympy/core/basic.py

```diff
@@ -122,6 +121,17 @@ def __new__(cls, *args):
         obj = object.__new__(cls)
         obj._assumptions = cls.default_assumptions
         obj._mhash = None  # will be set by __hash__ method.
+        for i,arg in enumerate(args):
```

Comment:
The `enumerate` here doesn't seem to be necessary. This could also just be written as 

```py
if not any(isinstance(arg, (Basic, ManagedProperties)) for arg in args):
    SymPyDeprecationWarning(...).warn()
```